exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -27,13 +27,104 @@ const (
 	ManagedControlPlaneFinalizer = "gcpmanagedcontrolplane.infrastructure.cluster.x-k8s.io"
 )
 
+// PrivateCluster defines a private Cluster.
+type PrivateCluster struct {
+	// EnablePrivateEndpoint: Whether the master's internal IP
+	// address is used as the cluster endpoint.
+	// +optional
+	EnablePrivateEndpoint bool `json:"enablePrivateEndpoint,omitempty"`
+
+	// EnablePrivateNodes: Whether nodes have internal IP
+	// addresses only. If enabled, all nodes are given only RFC
+	// 1918 private addresses and communicate with the master via
+	// private networking.
+	// +optional
+	EnablePrivateNodes bool `json:"enablePrivateNodes,omitempty"`
+
+	// ControlPlaneCidrBlock is the IP range in CIDR notation to use for the hosted master network. This range must not
+	// overlap with any other ranges in use within the cluster's network. Honored when enabled is true.
+	// +optional
+	ControlPlaneCidrBlock string `json:"controlPlaneCidrBlock,omitempty"`
+
+	// ControlPlaneGlobalAccess is whenever master is accessible globally or not. Honored when enabled is true.
+	// +optional
+	ControlPlaneGlobalAccess bool `json:"controlPlaneGlobalAccess,omitempty"`
+
+	// DisableDefaultSNAT disables cluster default sNAT rules. Honored when enabled is true.
+	// +optional
+	DisableDefaultSNAT bool `json:"disableDefaultSNAT,omitempty"`
+}
+
+// ClusterNetworkPod the range of CIDRBlock list from where it gets the IP address.
+type ClusterNetworkPod struct {
+	// CidrBlock is where all pods in the cluster are assigned an IP address from this range. Enter a range
+	// (in CIDR notation) within a network range, a mask, or leave this field blank to use a default range.
+	// This setting is permanent.
+	// +optional
+	CidrBlock string `json:"cidrBlock,omitempty"`
+}
+
+// ClusterNetworkService defines the range of CIDRBlock list from where it gets the IP address.
+type ClusterNetworkService struct {
+	// CidrBlock is where cluster services will be assigned an IP address from this IP address range. Enter a range
+	// (in CIDR notation) within a network range, a mask, or leave this field blank to use a default range.
+	// This setting is permanent.
+	// +optional
+	CidrBlock string `json:"cidrBlock,omitempty"`
+}
+
+// ClusterNetwork define the cluster network.
+type ClusterNetwork struct {
+	// PrivateCluster defines the private cluster spec.
+	// +optional
+	PrivateCluster *PrivateCluster `json:"privateCluster,omitempty"`
+
+	// UseIPAliases is whether alias IPs will be used for pod IPs in the cluster. If false, routes will be used for
+	// pod IPs in the cluster.
+	// +optional
+	UseIPAliases bool `json:"useIPAliases,omitempty"`
+
+	// Pod defines the range of CIDRBlock list from where it gets the IP address.
+	// +optional
+	Pod *ClusterNetworkPod `json:"pod,omitempty"`
```

Comment:
We don't use this field in this PR ... do we?

Reply:
Exactly, I set those expecting them to work. those don't do anything.